From 734efc5660349e774282b186a7597c46460d3f02 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Tue, 21 Mar 2023 22:28:32 -0300
Subject: [PATCH 4/4] GENIO: mt8195: board: use fixed fw_images definition

Prefer defining the known and fixed fw_images definition for capsule
updates.

Images are known (bl2 and fip), so define as part of fw_images
definition.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 board/mediatek/mt8195/mt8195_demo.c | 74 ++++++-----------------------
 include/configs/mt8195.h            | 18 +------
 2 files changed, 16 insertions(+), 76 deletions(-)

diff --git a/board/mediatek/mt8195/mt8195_demo.c b/board/mediatek/mt8195/mt8195_demo.c
index 2fb0558e3b5..3c50b47facc 100644
--- a/board/mediatek/mt8195/mt8195_demo.c
+++ b/board/mediatek/mt8195/mt8195_demo.c
@@ -11,10 +11,21 @@
 #include <asm/io.h>
 #include <linux/kernel.h>
 
-#define MT8195_UPDATABLE_IMAGES	2
-
 #if CONFIG_IS_ENABLED(EFI_HAVE_CAPSULE_SUPPORT)
-static struct efi_fw_image fw_images[MT8195_UPDATABLE_IMAGES] = {0};
+static struct efi_fw_image fw_images[] = {
+#if defined(CONFIG_EFI_CAPSULE_FIRMWARE_RAW)
+	{
+		.image_type_id = MT8195_DEMO_BL2_IMAGE_GUID,
+		.fw_name = u"MT8195-DEMO-BL2",
+		.image_index = 1,
+	},
+	{
+		.image_type_id = MT8195_DEMO_FIP_IMAGE_GUID,
+		.fw_name = u"MT8195-DEMO-FIP",
+		.image_index = 2,
+	},
+#endif
+};
 
 struct efi_capsule_update_info update_info = {
 	.dfu_string = "mmc 0=bl2.img raw 0x0 0x100 mmcpart 1;"
@@ -22,64 +33,9 @@ struct efi_capsule_update_info update_info = {
 	.images = fw_images,
 };
 
-u8 num_image_type_guids = MT8195_UPDATABLE_IMAGES;
+u8 num_image_type_guids = ARRAY_SIZE(fw_images);
 #endif
 
-#if defined(CONFIG_EFI_HAVE_CAPSULE_SUPPORT) && defined(CONFIG_EFI_PARTITION)
-static bool board_is_mt8195_demo(void)
-{
-	return CONFIG_IS_ENABLED(TARGET_MT8195) &&
-		of_machine_is_compatible("mediatek,mt8195-demo");
-}
-
-static bool board_is_genio_1200_evk(void)
-{
-	return CONFIG_IS_ENABLED(TARGET_MT8195) &&
-		of_machine_is_compatible("mediatek,genio-1200-evk");
-}
-
-static bool board_is_genio_1200_evk_ufs(void)
-{
-	return CONFIG_IS_ENABLED(TARGET_MT8195) &&
-		of_machine_is_compatible("mediatek,genio-1200-evk-ufs");
-}
-
-void mediatek_capsule_update_board_setup(void)
-{
-	if (board_is_mt8195_demo()) {
-		efi_guid_t image_type_guid =
-			MT8195_DEMO_FIT_IMAGE_GUID;
-		efi_guid_t uboot_image_type_guid = MT8195_DEMO_FIP_IMAGE_GUID;
-
-		guidcpy(&fw_images[0].image_type_id, &image_type_guid);
-		guidcpy(&fw_images[1].image_type_id, &uboot_image_type_guid);
-
-		fw_images[0].fw_name = u"MT8195-DEMO-FIT";
-		fw_images[1].fw_name = u"MT8195-DEMO-FIP";
-	} else if (board_is_genio_1200_evk()) {
-		efi_guid_t image_type_guid =
-			GENIO_1200_EVK_FIT_IMAGE_GUID;
-		efi_guid_t uboot_image_type_guid = GENIO_1200_EVK_FIP_IMAGE_GUID;
-
-		guidcpy(&fw_images[0].image_type_id, &image_type_guid);
-		guidcpy(&fw_images[1].image_type_id, &uboot_image_type_guid);
-
-		fw_images[0].fw_name = u"GENIO-1200-EVK-FIT";
-		fw_images[1].fw_name = u"GENIO-1200-EVK-FIP";
-	} else if (board_is_genio_1200_evk_ufs()) {
-		efi_guid_t image_type_guid =
-			GENIO_1200_EVK_UFS_FIT_IMAGE_GUID;
-		efi_guid_t uboot_image_type_guid = GENIO_1200_EVK_UFS_FIP_IMAGE_GUID;
-
-		guidcpy(&fw_images[0].image_type_id, &image_type_guid);
-		guidcpy(&fw_images[1].image_type_id, &uboot_image_type_guid);
-
-		fw_images[0].fw_name = u"GENIO-1200-EVK-UFS-FIT";
-		fw_images[1].fw_name = u"GENIO-1200-EVK-UFS-FIP";
-	}
-}
-#endif /* CONFIG_EFI_HAVE_CAPSULE_SUPPORT && CONFIG_EFI_PARTITION */
-
 int board_init(void)
 {
 	struct udevice *dev;
diff --git a/include/configs/mt8195.h b/include/configs/mt8195.h
index e753224cf02..a4aaa30f39a 100644
--- a/include/configs/mt8195.h
+++ b/include/configs/mt8195.h
@@ -17,7 +17,7 @@
 #define CONFIG_SYS_NS16550_COM1		0x11002000
 #define CONFIG_SYS_NS16550_CLK		26000000
 
-#define MT8195_DEMO_FIT_IMAGE_GUID \
+#define MT8195_DEMO_BL2_IMAGE_GUID \
 	EFI_GUID(0x0538fd5d, 0x00dc, 0x47c5, 0x8d, 0x26, \
 		 0x43, 0x48, 0x03, 0xb1, 0xe3, 0x27)
 
@@ -25,22 +25,6 @@
 	EFI_GUID(0xf65ec70e, 0x45ab, 0x40fb, 0x99, 0x52, \
 		 0xe5, 0xbc, 0x30, 0x70, 0xda, 0x5c)
 
-#define GENIO_1200_EVK_FIT_IMAGE_GUID \
-	EFI_GUID(0x9fd30648, 0xb128, 0x440d, 0xbd, 0xd9, \
-		 0x0e, 0x53, 0xeb, 0x2a, 0x3c, 0xb8)
-
-#define GENIO_1200_EVK_FIP_IMAGE_GUID \
-	EFI_GUID(0x39961e72, 0x5a8e, 0x445a, 0x90, 0xfe, \
-		 0xed, 0x68, 0x33, 0x07, 0x44, 0xec)
-
-#define GENIO_1200_EVK_UFS_FIT_IMAGE_GUID \
-	EFI_GUID(0x49be7238, 0x9c9f, 0x4e4f, 0x94, 0x45, \
-		 0x38, 0x9f, 0xcf, 0xd2, 0xa5, 0x12)
-
-#define GENIO_1200_EVK_UFS_FIP_IMAGE_GUID \
-	EFI_GUID(0x0c1603a7, 0x7f3d, 0x427c, 0x9e, 0xcc, \
-		 0xf9, 0xb0, 0x2d, 0x2f, 0xc7, 0x29)
-
 /* Environment settings */
 #include <config_distro_bootcmd.h>
 
-- 
2.39.1

